test/suite/test_hs20.py

```diff
@@ -32,8 +32,14 @@
 # test_hs20.py
 # Ensure we never reconstruct a reverse modify update in the history store based on the onpage overflow value
 class test_hs20(wttest.WiredTigerTestCase):
+    # This test purposely uses a small cache to force eviction to occur
+    # at certain points in its execution.
     conn_config = 'cache_size=50MB,eviction=(threads_max=1)'
 
+    # The small cache size of the test means that rollbacks will occur.
+    # Add more retries to make eventual success more likely.
+    rollbacks_allowed = 5
```

Comment:
I like the general change and the fact that setting this here makes the retry hooks underneath work well. Your change here itself does not have an issue.

I have a general concern that, similar to the sleep/timing type tests, that we'll still get arbitrary failures when 5 retries isn't quite enough or some systems may make it more prone than others, etc. Is the number of retries going to help (as in maybe it allows more time for internal eviction to allow it to pass on the next try)?

Reply:
It's hard to know the circumstances that allow the test to pass sometimes but not others.  A retry, in this case, restarts the test entirely.  The old connection is closed and a new one open.  So time for internal eviction is not really the issue.  We do know that the test generally passes - or we'd see a lot more failures.  And at least for the failure in the ticket, we know it did roll back three times in a row.

So yeah, I agree that it's hard to know if this will really fix it, but I'd like to give it a shot.